auth0/v3/management/rest.py

```diff
@@ -152,7 +160,7 @@ def _error_message(self):
 
 class EmptyResponse(Response):
     def __init__(self, status_code):
-        super(EmptyResponse, self).__init__(status_code, '')
+        super(EmptyResponse, self).__init__(status_code, '', {})
```

Comment:
Do you not expect the API endpoints that return empty responses to get rate limited?

Reply:
Ah, I see - so if there's no `response.text` we know it's been successful so no need to check the rate limit header?

Reply:
Responses sent from the server could be:
- 204 NO CONTENT (empty body, empty text)
- Plain text response: e.g. "User is blocked"
- JSON response: e.g. { "code": "success" }

Every Response instance of this SDK defines a method to check if they represent errors, by checking against the status code here 
https://github.com/auth0/auth0-python/blob/master/auth0/v3/management/rest.py#L111-L112. 

Since for now, the headers are only being used for rate limit purposes, and only in the case of an errored response, I think it makes sense to pass an empty dict here. In addition, this `headers` property in the Response is not accessible for developers since the parsed content is what is being returned after a successful call, or an exception is raised if this represents an error.

https://github.com/auth0/auth0-python/blob/master/auth0/v3/management/rest.py#L84-L87

Reply:
replying to your question, 429 responses have been returning a non-empty body. I don't expect empty responses to represent a rate limit exception ever. 👍